pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -456,49 +456,7 @@ func buildGatewayListenerTLSContext(
 			}
 		}
 	} else if server.Tls.Mode == networking.Server_TLSOptions_ISTIO_MUTUAL {
-		// global SDS must be enabled if the sds path is non-default, configure TLS with SDS
-		if sdsPath != "" {
-			// configure egress with SDS
-			tls.CommonTlsContext.ValidationContextType = &auth.CommonTlsContext_CombinedValidationContext{
-				CombinedValidationContext: &auth.CommonTlsContext_CombinedCertificateValidationContext{
-					DefaultValidationContext:         &auth.CertificateValidationContext{VerifySubjectAltName: server.Tls.SubjectAltNames},
-					ValidationContextSdsSecretConfig: authn_model.ConstructSdsSecretConfig(authn_model.SDSRootResourceName, sdsPath),
-				},
-			}
-			tls.CommonTlsContext.TlsCertificateSdsSecretConfigs = []*auth.SdsSecretConfig{
-				authn_model.ConstructSdsSecretConfig(authn_model.SDSDefaultResourceName, sdsPath)}
-		} else {
-			// global SDS disabled, fall back on using mounted certificates
-			caCertificates := model.GetOrDefault(metadata.TLSServerRootCert, constants.DefaultRootCert)
-			trustedCa := &core.DataSource{
-				Specifier: &core.DataSource_Filename{
-					Filename: caCertificates,
-				},
-			}
-			tls.CommonTlsContext.ValidationContextType = &auth.CommonTlsContext_ValidationContext{
-				ValidationContext: &auth.CertificateValidationContext{
-					TrustedCa:            trustedCa,
-					VerifySubjectAltName: server.Tls.SubjectAltNames,
-				},
-			}
-
-			certChainPath := model.GetOrDefault(metadata.TLSServerCertChain, constants.DefaultCertChain)
-			privateKeyPath := model.GetOrDefault(metadata.TLSServerKey, constants.DefaultKey)
-			tls.CommonTlsContext.TlsCertificates = []*auth.TlsCertificate{
-				{
-					CertificateChain: &core.DataSource{
-						Specifier: &core.DataSource_Filename{
-							Filename: certChainPath,
-						},
-					},
-					PrivateKey: &core.DataSource{
-						Specifier: &core.DataSource_Filename{
-							Filename: privateKeyPath,
-						},
-					},
-				},
-			}
-		}
+		authn_utils.ApplyToCommonTLSContext(tls.CommonTlsContext, metadata, sdsPath, server.Tls.SubjectAltNames)
```

Comment:
Do you think we should move this code to networking/util folder to avoid any circular dependencies in future? both authn and networking/core can import this. Otherwise, we will also have a bunch of util packages everywhere.

Reply:
I'm ok with refactor the code location with proper ownership.

Reply:
OK, will do that